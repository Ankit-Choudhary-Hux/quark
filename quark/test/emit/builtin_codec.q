String hexb(byte v) {
    Codec c = defaultCodec();
    Buffer b = c.buffer(1);
    b.putByte(0,v);
    return c.toHexdump(b,0,1,10);
}

String hexs(short v) {
    Codec c = defaultCodec();
    Buffer b = c.buffer(2);
    b.putShort(0,v);
    return c.toHexdump(b,0,2,10);
}

String hexi(int v) {
    Codec c = defaultCodec();
    Buffer b = c.buffer(4);
    b.putInt(0,v);
    return c.toHexdump(b,0,4,10);
}

String hexl(long v) {
    Codec c = defaultCodec();
    Buffer b = c.buffer(8);
    b.putLong(0,v);
    return c.toHexdump(b,0,8,10);
}

// xfail:py
// xfail:js

void main() {
    Codec c = defaultCodec();
    String hello = "hello world!";
    Buffer b = c.buffer(500);
    print(b.putStringUTF8(0, hello).toString());
    print(c.toHexdump(b, 0, hello.size(), 2));
    print(c.toBase64(b, 0, hello.size()));

    Buffer d = c.fromHexdump("0x0102 0304 0506 0708 09");
    print(c.toHexdump(d, 0, d.capacity(), 1));
    int i = 0;
    int e = d.capacity();
    while (i < e) {
        if ( (i + 1) < (e + 1)) {
            byte vb = d.getByte(i);
            print("B[" + i.toString() + "]=" + vb.toString() + "   " + hexb(vb));
        }
        if ( (i + 2) < (e + 1)) {
            short vs = d.getShort(i);
            print("S[" + i.toString() + "]=" + vs.toString() + "   " + hexs(vs));
        }
        if ( (i + 4) < (e + 1)) {
            int vi = d.getInt(i);
            print("I[" + i.toString() + "]=" + vi.toString() + "   " + hexi(vi));
        }
        if ( (i + 8) < (e + 1)) {
            long vl = d.getLong(i);
            print("L[" + i.toString() + "]=" + vl.toString() + "   " + hexl(vl));
        }
        i = i + 1;
    }

    Buffer f = c.fromBase64("eWF5");
    print(f.getStringUTF8(0,3));
}
