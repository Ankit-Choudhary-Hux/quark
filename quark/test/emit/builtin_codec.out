12
68656c6c 6f20776f 726c6421
aGVsbG8gd29ybGQh
0000 0001 0203 0405 06
B[0]=0   00
S[0]=0   0000
I[0]=1   00000001
L[0]=4328719365   0000000102030405
B[1]=0   00
S[1]=0   0000
I[1]=258   00000102
L[1]=1108152157446   0000010203040506
B[2]=0   00
S[2]=1   0001
I[2]=66051   00010203
B[3]=1   01
S[3]=258   0102
I[3]=16909060   01020304
B[4]=2   02
S[4]=515   0203
I[4]=33752069   02030405
B[5]=3   03
S[5]=772   0304
I[5]=50595078   03040506
B[6]=4   04
S[6]=1029   0405
B[7]=5   05
S[7]=1286   0506
B[8]=6   06
yay
